python/lsst/ctrl/execute/allocator.py

```diff
@@ -221,6 +219,8 @@ def createSubmitFile(self, inputFile):
         outfile : `str`
             The newly created file name
         """
+        if not os.path.exists(self.configDir):
+            os.makedirs(self.configDir)
```

Comment:
Happened to check out function docstring so I could understand where the code moved to.  "PBS" is outdated, right?  Also there should be a Parameters section in the docstring for inputFile

Reply:
Could the makedirs be moved to inside createFile (i.e., makes sure path exists for destination file before writing)?  That way the order of the createFile calls won't matter (right now createSubmitFile has to be done first because it has the makedirs).  Also os.makedirs has flag to ignore errors if already exists (https://docs.python.org/3/library/os.html#os.makedirs)

Reply:
I think this function would be used for both Slurm and PBS if we did work to support both, so I will change it to "batch submit file".

Reply:
Optimal placement for directory and file creation will probably take more study/testing in another ticket.